src/phoenix/experimental/evals/models/openai.py

```diff
@@ -90,51 +128,77 @@ def _init_environment(self) -> None:
             )
 
     def _init_open_ai(self) -> None:
+        # For Azure, you need to provide the endpoint and the endpoint
+        self._is_azure = bool(self.azure_endpoint)
+
         self._model_uses_legacy_completion_api = self.model_name.startswith(
             LEGACY_COMPLETION_API_MODELS
         )
-        if self.openai_api_key is None:
+        if self.api_key is None:
             api_key = os.getenv(OPENAI_API_KEY_ENVVAR_NAME)
             if api_key is None:
                 # TODO: Create custom AuthenticationError
                 raise RuntimeError(
-                    "OpenAI's API key not provided. Pass it as an argument to 'openai_api_key' "
+                    "OpenAI's API key not provided. Pass it as an argument to 'api_key' "
                     "or set it in your environment: 'export OPENAI_API_KEY=sk-****'"
                 )
-            self.openai_api_key = api_key
-        self._client = self._openai.Client(api_key=self.openai_api_key)
-        self.openai_api_type = self.openai_api_type or self._openai.api_type
-        self.openai_api_version = self.openai_api_version or self._openai.api_version
-        self.openai_organization = self.openai_organization or self._openai.organization
+            self.api_key = api_key
 
-        self._is_azure = (self._openai.api_type or "").lower().startswith("azure")
+        # Set the version, organization, and base_url - default to openAI
+        self.api_version = self.api_version or self._openai.api_version
+        self.organization = self.organization or self._openai.organization
 
+        # Initialize specific clients depending on the API backend
+        # Set the type first
+        self._client: Union[self._openai.OpenAI, self._openai.AzureOpenAI]  # type: ignore
         if self._is_azure:
-            if not self.engine:
-                raise ValueError(
-                    "You must provide the deployment name in the 'engine' parameter "
-                    "to access the Azure OpenAI service"
-                )
-            self._openai_api_model_name = self.engine
-        elif self.model_name in MODEL_TOKEN_LIMIT_MAPPING.keys():
-            self._openai_api_model_name = self.model_name
-        elif "gpt-3.5-turbo" in self.model_name:
-            self._openai_api_model_name = "gpt-3.5-turbo-0613"
-        elif "gpt-4" in self.model_name:
-            self._openai_api_model_name = "gpt-4-0613"
-        else:
-            raise NotImplementedError(
-                f"openai_api_model_name not available for model {self.model_name}. "
```

Comment:
I'm nixing this largely confusing block. I have no idea why this flattening has to happen and starting to feel this tiktoken business is not worth maintaining.